runtime/doc/lua.txt

```diff
@@ -2144,4 +2144,102 @@ set({mode}, {lhs}, {rhs}, {opts})                           *vim.keymap.set()*
                 See also: ~
                     |nvim_set_keymap()|
 
+
+==============================================================================
+Lua module: path                                                    *lua-path*
+
+basename({file})                                         *vim.path.basename()*
+                Return the basename of the given file or directory
+
+                Parameters: ~
+                    {file}  (string) File or directory
+
+                Return: ~
+                    (string) Basename of {file}
+
+dir({path})                                                   *vim.path.dir()*
+                Return an iterator over the files and directories located in
```

Comment:
The iterator provides the file or directory basename or full path?

Reply:
Each file/directory name returned from the iterator is relative to the provided `{path}` argument. I added a note about this to the docs.